Shared/Samples/Show device location using indoor positioning/ShowDeviceLocationUsingIndoorPositioningView.Model.swift

```diff
@@ -0,0 +1,226 @@
+        /// The location manager which handles the location data.
+        private let locationManager = CLLocationManager()
+        
+        /// The measurement formatter for sensor accuracy.
+        private let measurementFormatter: MeasurementFormatter = {
+            let formatter = MeasurementFormatter()
+            formatter.unitStyle = .short
+            formatter.unitOptions = .providedUnit
+            formatter.numberFormatter.maximumFractionDigits = 1
+            formatter.numberFormatter.roundingMode = .up
+            return formatter
+        }()
+        
+        /// Kicks off the logic for displaying the indoors position.
+        func displayIndoorData() async throws {
+            try await setIndoorDatasource()
+            try await requestLocationServicesAuthorizationIfNecessary()
+        }
+        
+        /// Stops the location data source.
+        func stopLocationDataSource() {
+            Task {
+                await locationDisplay.dataSource.stop()
+            }
+        }
+        
+        /// A function that attempts to load an indoor definition attached to the map
+        /// and returns a boolean value based whether it is loaded.
+        /// - Parameter map: The map that contains the IndoorDefinition.
+        /// - Returns: A boolean value for whether the IndoorDefinition is loaded.
+        private func indoorDefinitionIsLoaded(map: Map) async throws -> Bool {
+            guard map.indoorPositioningDefinition?.loadStatus != .loaded else { return true }
+            try await map.indoorPositioningDefinition?.load()
+            return map.indoorPositioningDefinition?.loadStatus == .loaded
+        }
+        
+        /// Sets the indoor datasource on the location display depending on
+        /// whether the map contains an IndoorDefinition.
+        private func setIndoorDatasource() async throws {
+            try await map.floorManager?.load()
+            // If an indoor definition exists in the map, it gets loaded and sets the IndoorsDataSource to pull information
+            // from the definition.
+            if try await indoorDefinitionIsLoaded(map: map),
+               let indoorPositioningDefinition = map.indoorPositioningDefinition {
+                indoorsLocationDataSource = IndoorsLocationDataSource(definition: indoorPositioningDefinition)
+            // Otherwise the IndoorsDataSource attempts to create itself using IPS table information.
+            } else {
+                indoorsLocationDataSource = try await createIndoorLocationDataSource(map: map)
+            }
+            // This ensures that the details of the inside of the building, like room layouts are displayed.
+            for featLayer in map.operationalLayers {
+                if featLayer.name == "Transitions" || featLayer.name == "Details" {
+                    featLayer.isVisible = true
+                }
+            }
+            guard let dataSource = indoorsLocationDataSource else { return }
```

Comment:
Add comment that the datasource should always be there, but createIndoorLocationDataSource throws an optional value so it cannot be guaranteed. Best option is you get to this point without a datasource is to return (ideally an error would have been thrown before this point and the flow broken.